### PR TITLE
Add Newton-Ramson cube root approximation

### PR DESCRIPTION
**Pull request by rostin79s**
# New Nada Example

- [x] I have read the [contributing docs](/NillionNetwork/nada-by-example/blob/main/CONTRIBUTING.md)
- [x] This is not a duplicate of any [existing pull request](https://github.com/NillionNetwork/nada-by-example/pulls)

## Description

Using Newtons method to approximate the cube root function.  Function takes input `x` and returns integer cube root value `s`.

The number of loop iterations for this method should be 11 for 8-bit numbers, 22 for 16-bit numbers and 41 for 32-bit numbers. It is currently set to 41. 

## Use Case

Polynomial evaluation.

## Related Issues

The function overflows for large integers, and will return incorrect results.
```

## Optional: add your information for a Nada by Example Contributor POAP

Your Twitter:
Your Ethereum address:

### CODE REVIEW

File: tests/cube_root_test.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+program: cube_root
+inputs:
+  num:
+    SecretInteger: "127324"
+expected_outputs:
+  my_output:
+    SecretInteger: "1"
```
Comment (jimouris) [inline tests/cube_root_test.yaml:8]: does this mean that the cube root of `127324` is `1`?

**Comment by rostin79s** [comment]
That's a mistake, it should be 50.

File: src/cube_root.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+"""Implementation of the square root function using the Newton method
```
Comment (jimouris) [inline src/cube_root.py:1]:
```suggestion
"""Implementation of the cube root function using the Newton method
```

**Review by jimouris — APPROVED** [review]
Nice!

**Review by oceans404 — APPROVED** [review]
Great work on this! Thanks for contributing cube root

**Review by Davetbutler — APPROVED** [review]
Nice one. Looking forward to continue this mini colab :)

**Pull request merged**